plugins/distance_table.hpp

```diff
@@ -119,17 +157,41 @@ template <class DataFacadeT> class DistanceTablePlugin final : public BasePlugin
             return 400;
         }
 
-        osrm::json::Array json_array;
-        const auto number_of_locations = phantom_node_vector.size();
-        for (const auto row : osrm::irange<std::size_t>(0, number_of_locations))
+        osrm::json::Array matrix_json_array;
+        const auto number_of_targets = phantom_node_target_vector.size();
+        const auto number_of_sources = (phantom_node_source_vector.size()) ? phantom_node_source_vector.size() : number_of_targets;
+        for (const auto row : osrm::irange<std::size_t>(0, number_of_sources))
         {
             osrm::json::Array json_row;
-            auto row_begin_iterator = result_table->begin() + (row * number_of_locations);
-            auto row_end_iterator = result_table->begin() + ((row + 1) * number_of_locations);
+            auto row_begin_iterator = result_table->begin() + (row * number_of_targets);
+            auto row_end_iterator = result_table->begin() + ((row + 1) * number_of_targets);
             json_row.values.insert(json_row.values.end(), row_begin_iterator, row_end_iterator);
-            json_array.values.push_back(json_row);
+            matrix_json_array.values.push_back(json_row);
+        }
+        json_result.values["distance_table"] = matrix_json_array;
+        if (route_parameters.mapped_points) {
```

Comment:
Remove if, since the parameters needs to be dropped. (Code in general is fine.) I would go with `source_coordinates` and `target_coordinates` here.